File: Week6/sentimental-mario-more/mario.py
# Import get_int from cs50 library
from cs50 import get_int

# Define main function


def main():
    # Ask user for a valid pyramid height bewteen 1 and 8
    while True:
        height = get_int("Input pyramid height, between 1 and 8: ")
        if height > 0 and height < 9:
            break
    # Call on pyramid function to print
    pyramid(height)

# Print the pyramid


def pyramid(height):
    for i in range(height):
        for j in range(height - i - 1):
            print(" ", end='')
        for j in range(i + 1):
            print("#", end='')
        print("  ", end='')
        for j in range(i + 1):
            print("#", end='')
        print("\n", end='')


# Call main function
main()

File: Week1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height = 0;

    // Ask the user for input and loop until a valid number between 1 and 8 has been provided
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Create a loop that determines the number of rows based on input
    for (int i = 0; i < height; i++)
    {
        // Print height - row spaces - 1
        for (int j = 0; j < height - i - 1; j++)
        {
            printf(" ");
        }
        // Print row number of hashes, as measured by k
        for (int k = 0; k <= i; k ++)
        {
            printf("#");
        }
        // Print a new line
        printf("\n");
    } 
}